src/common/ceph_context.cc

```diff
@@ -569,6 +578,17 @@ PerfCountersCollection *CephContext::get_perfcounters_collection()
   return _perf_counters_collection;
 }
 
+void CephContext::enable_perf_counter()
+{
+  ceph_spin_lock(&_cct_perf_lock);
```

Comment:
any specific reason to use a spinlock here instead of a mutex?  granted it shouldn't be contended, but ..

actually, i'm not sure the spinlock is even needed.  the perfcounter is thread safe, and if you don't assign the pointer until it is ready to be used, then the service thread doesn't need a lock.  you just need to make sure you shut down the thread before shutting down the perfcounter.